packages/amplify-auth-sdk/src/stores/secure-store.js

```diff
@@ -19,19 +25,64 @@ export default class SecureStore extends FileStore {
 	/**
 	 * Loads the `keytar` library and initializes the token file.
 	 *
-	 * @param {Object} [opts] - Various options.
+	 * @param {Object} opts - Various options.
+	 * @param {String} opts.homeDir - The path to the home directory containing the `lib`
+	 * directory where `keytar` is located.
 	 * @param {String} [opts.secureServiceName="Axway AMPLIFY Auth"] - The name of the consumer
      * using this library.
 	 * @access public
 	 */
 	constructor(opts = {}) {
+		if (!opts || typeof opts !== 'object') {
+			throw E.INVALID_ARGUMENT('Expected opts to be an object');
+		}
+
+		const { homeDir } = opts;
+		if (!homeDir || typeof homeDir !== 'string') {
+			throw E.INVALID_PARAMETER('Secure store requires the home directory to be specified');
+		}
+
+		const prefix = path.join(homeDir, 'lib', 'keytar', `${process.platform}_${process.arch}_${process.versions.modules}`);
+		const keytarPath = path.join(prefix, 'node_modules', 'keytar');
+		log(`Attempting to load keytar: ${highlight(keytarPath)}`);
+
 		let keytar;
 
 		try {
-			keytar = require('keytar');
+			keytar = require(keytarPath);
 		} catch (e) {
-			/* istanbul ignore next */
-			throw E.KEYTAR_NOT_FOUND('"keytar" package not found, is it installed?');
+			if (e.code !== 'MODULE_NOT_FOUND') {
+				log(e);
+				throw e;
+			}
+
+			const args = [ 'install', 'keytar@4.4.1', '--no-save', '--production', '--prefix', prefix ];
+			log(`keytar not found, running: ${highlight(`npm ${args.join(' ')}`)}`);
+			fs.removeSync(keytarPath);
+
+			const env = Object.assign({ NO_UPDATE_NOTIFIER: 1 }, process.env);
+			const result = spawnSync('npm', args, { env });
```

Comment:
I think you'll need `shell: true` in the options for spawning npm on windows

Reply:
Fixed!